packages/block-editor/src/components/block-list/block.js

```diff
@@ -221,7 +232,7 @@ function BlockListBlock( {
 					isTemporarilyEditingAsBlocks,
 			},
 			dataAlign && themeSupportsLayout && `align${ dataAlign }`,
-			className
+			! ( dataAlign && isSticky ) && className
```

Comment:
Probably not an issue, but if there are other block supports that generate a className (like element colors for link colors) then these classnames will also be moved up to the align wrapper if the block has been set to sticky? I think this is most likely fine, though, as we're already moving those classnames that need to act on the _inner_ wrapper down to the inner wrapper to handle block gap, so it's all working nicely for me locally.

![image](https://github.com/WordPress/gutenberg/assets/14988353/4918b306-0248-4552-8d41-58722e832c6d)

I.e. I can both set to sticky _and_ set a custom block gap 👍

Reply:
Hmmm I tried a few supports in testing and the others didn't seem to be included in the `className` variable. I'll check again. Also now I'm wondering if we can leverage classname swapping to fix the issue with background color overlapping background image too:

<img width="942" alt="Screenshot 2023-12-05 at 3 01 49 pm" src="https://github.com/WordPress/gutenberg/assets/8096000/8c6450e7-bea5-4739-bdb9-9125a279a99c">

Reply:
So the only other classname being transferred to the alignment wrapper is `wp-elements-x`, which isn't problematic because its styles don't target direct descendants. If this did become a problem later on we could perhaps filter out the position classnames, but best avoid adding extra logic if it isn't needed. 

What I did notice is that in the editor, when testing with a single group block on the page. position and layout classnames were identical `wp-container-0`. In practice that can't cause problems because applying sticky position to the inner group container won't do anything, and neither will the block spacing rules override margins set on the block. To be on the safe side though, we might want to consider changing the layout class in the editor to match the front end `wp-container-core-group-layout-x` (and possibly a similar specificity could be applied to the position class at some point). I'd do that as a separate PR though.

Reply:
Update: the background styling in the screenshot above is specific to TT1, so I guess it's fine to leave as is. Imo it's not worth the overhead of trying to attach the background image to the inner container, because it's pretty easy to fix this in the UI.

Reply:
Sounds good. The less we need to mutate for the classic themes case the better, if possible. We can always continue exploring in follow-ups if need be, too!